lib/samples/show_viewshed_from_geoelement_in_scene/show_viewshed_from_geoelement_in_scene.dart

```diff
@@ -0,0 +1,288 @@
+      if (distance <= 5) {
+        _waypoint = null;
+        timer.cancel();
+        return;
+      }
+
+      // Move a small step toward the waypoint
+      const step = 1.0; // meters
+      final movedPoints = GeometryEngine.moveGeodetic(
+        pointCollection: [currentPos],
+        distance: step,
+        azimuth: azimuth,
+        distanceUnit: LinearUnit(unitId: LinearUnitId.meters),
+        azimuthUnit: AngularUnit(unitId: AngularUnitId.degrees),
+        curveType: GeodeticCurveType.geodesic,
+      );
+
+      if (movedPoints.isEmpty) return;
+
+      final newPoint = movedPoints.first;
+      _tankGraphic!.geometry = newPoint;
+
+      // Update heading
+      final currentHeading =
+          (_tankGraphic!.attributes['HEADING'] as num?)?.toDouble() ?? 0.0;
+
+      final headingDiff = shortestAngle(currentHeading, azimuth);
+      final newHeading = currentHeading + headingDiff / 10;
+
+      _tankGraphic!.attributes['HEADING'] = newHeading;
+    });
+  }
+
+  // Calculate shortest angle to rotate
+
+  double shortestAngle(double from, double to) {
+    final difference = (to - from + 540) % 360 - 180;
+    return difference;
+  }
+
+  // Loads the 3D tank model from local sample data and returns it as a Graphic.
+  Future<Graphic> _loadTankGraphic() async {
+    await downloadSampleData(['07d62a792ab6496d9b772a24efea45d0']);
+    final documentsDirPath =
+        (await getApplicationDocumentsDirectory()).absolute.path;
+    final tankModelPath = '$documentsDirPath/bradley_low_3ds/bradle.3ds';
+
+    // Define the tank symbol.
+    final tankSymbol =
+        ModelSceneSymbol.withUri(uri: Uri.parse(tankModelPath), scale: 10)
+          ..heading = 90
+          ..anchorPosition = SceneSymbolAnchorPosition.bottom;
+
+    // Return the graphic that combines geometry and symbol.
+    return Graphic(
+      geometry: ArcGISPoint(x: -4.506390, y: 48.385624),
+      attributes: {'HEADING': 0.0},
+      symbol: tankSymbol,
+    );
+  }
+
+  // Adds the tank graphic to a graphics overlay and sets the initial viewpoint.
+  Future<void> _addTankToScene(Graphic tankGraphic) async {
+    final graphicsOverlay = GraphicsOverlay()
+      ..graphics.add(tankGraphic)
+      ..sceneProperties = LayerSceneProperties(
+        surfacePlacement: SurfacePlacement.relative,
+      );
+
+    // Configure the heading expression for the tank; this will allow the
+    // viewshed to update automatically based on the tank's position.
+    final renderer = SimpleRenderer()
+      ..sceneProperties.headingExpression = '[HEADING]'
+      ..sceneProperties.pitchExpression = '[PITCH]'
+      ..sceneProperties.rollExpression = '[ROLL]';
+    graphicsOverlay.renderer = renderer;
+
+    _sceneViewController.graphicsOverlays.add(graphicsOverlay);
+  }
+
+  // Add viewshed to the scene
+  Future<void> _addViewShedToScene(Graphic tankGraphic) async {
+    // Create a GeoElementViewShed attached to the scene.
+    final geoElementViewShed =
```

Comment:
In the comment and in the variable name, we should standardize the capitalization to use a lower case "s" in "Viewshed". (It's spelled "Viewshed" in our API.)